modules/vector-sets/expr.c

```diff
@@ -240,9 +235,10 @@ void exprConsumeSpaces(exprstate *es) {
     while(es->p[0] && isspace(es->p[0])) es->p++;
 }
 
-/* Parse an operator, trying to match the longer match in the
- * operators table. */
-exprtoken *exprParseOperator(exprstate *es) {
+/* Parse an operator or a literal (just "null" currently).
+ * When parsing operators, the functil will try to match the longest match
+ * in the operators table. */
+exprtoken *exprParseOperatorOrLiteral(exprstate *es) {
```

Comment:
Could we extract literal case handling? Or instead, could we remove "or" from the function name?

Reply:
I checked very carefully (and even implemented extracting) and the result code is worse. It requires to have a function that returns failure and another to be called instead. The functions right now never need to do backtracking restoring the pointer, now they would instead. Basically there is nothing wrong with the current implementation since operators and literal values like null, true, false are grammatically the same, and the distinction is only semantical, so it makes sense to handle them like it is now. If there was zero complexity cost maybe it could be worth splitting the case? Maybe (but with minimal gains). But not in this case. Thanks.

Reply:
All good, I get it. Thanks for investigating this.